packages/core/src/LernaTool.ts

```diff
@@ -0,0 +1,86 @@
+import path from 'path';
+import fs from 'fs-extra';
+
+import { Tool, ToolType, Package, PackageJSON, Packages, InvalidMonorepoError } from './Tool';
+import { expandPackageGlobs, expandPackageGlobsSync } from "./expandPackageGlobs";
+
+export interface LernaJson {
+    useWorkspaces?: boolean;
+    packages?: string[];
+}
+
+export const LernaTool: Tool = {
+    type: 'lerna',
+
+    async isMonorepoRoot(directory: string): Promise<boolean> {
+        try {
+            const lernaJson = await fs.readJson(path.join(directory, "lerna.json")) as LernaJson;
+            if (lernaJson.useWorkspaces !== true) {
+                return true;
+            }
+        } catch (err) {
+            if (err.code !== "ENOENT") {
+                throw err;
+            }
+        }
+        return false;
+    },
+
+    isMonorepoRootSync(directory: string): boolean {
+        try {
+            const lernaJson = fs.readJsonSync(path.join(directory, "lerna.json")) as LernaJson;
+            if (lernaJson.useWorkspaces !== true) {
+                return true;
+            }
+        } catch (err) {
+            if (err.code !== "ENOENT") {
+                throw err;
+            }
+        }
+        return false;
+    },
+
+    async getPackages(directory: string): Promise<Packages> {
+        try {
+            const lernaJson = await fs.readJson(path.join(directory, "lerna.json"));
+            const pkgJson = (await fs.readJson(path.join(directory, "package.json"))) as PackageJSON;
+            const packageGlobs: string[] = lernaJson.packages || ["packages/*"];
+
+            return {
```

Comment:
In theory, we should also return `npmClient` here somehow (since Lerna might be configured with different clients). Nobody has ever asked for this though... so I'm totally OK with not doing that

Reply:
Very useful feedback here, but let me ask... let's say I'm using Lerna (my monorepo tool), and I'm possibly using one of npm, pnpm, or yarn (my package manager).

What difference would each of these make? Is the choice of tool relevant at all in versioning / changeset consumption / changelog updates, or is it basically at the end of the process the "publish" command only (npm publish vs pnpm publish vs yarn publish).

Reply:
It's sometimes relevant when it comes to publishing - not all package managers handle all of the "available" protocols (although I have a hard time figuring out a scenario in which this would matter for single package repos). The more important difference is that they might handle authentication differently.

But as mentioned, it's actually not a new problem so we can postpone fixing this.